types/basetypes/float32_type.go

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package basetypes
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"math/big"
+
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/internal/logging"
+)
+
+// Float32Typable extends attr.Type for float32 types.
+// Implement this interface to create a custom Float32Type type.
+type Float32Typable interface {
+	attr.Type
+
+	// ValueFromFloat32 should convert the Float32 to a Float32Valuable type.
+	ValueFromFloat32(context.Context, Float32Value) (Float32Valuable, diag.Diagnostics)
+}
+
+var _ Float32Typable = Float32Type{}
+
+// Float32Type is the base framework type for a floating point number.
+// Float32Value is the associated value type.
+type Float32Type struct{}
+
+// ApplyTerraform5AttributePathStep applies the given AttributePathStep to the
+// type.
+func (t Float32Type) ApplyTerraform5AttributePathStep(step tftypes.AttributePathStep) (interface{}, error) {
+	return nil, fmt.Errorf("cannot apply AttributePathStep %T to %s", step, t.String())
+}
+
+// Equal returns true if the given type is equivalent.
+func (t Float32Type) Equal(o attr.Type) bool {
+	_, ok := o.(Float32Type)
+
+	return ok
+}
+
+// String returns a human readable string of the type name.
+func (t Float32Type) String() string {
+	return "basetypes.Float32Type"
+}
+
+// TerraformType returns the tftypes.Type that should be used to represent this
+// framework type.
+func (t Float32Type) TerraformType(_ context.Context) tftypes.Type {
+	return tftypes.Number
+}
+
+// ValueFromFloat32 returns a Float32Valuable type given a Float32Value.
+func (t Float32Type) ValueFromFloat32(_ context.Context, v Float32Value) (Float32Valuable, diag.Diagnostics) {
+	return v, nil
+}
+
+// ValueFromTerraform returns a Value given a tftypes.Value.  This is meant to
+// convert the tftypes.Value into a more convenient Go type for the provider to
+// consume the data with.
+func (t Float32Type) ValueFromTerraform(ctx context.Context, in tftypes.Value) (attr.Value, error) {
+	if !in.IsKnown() {
+		return NewFloat32Unknown(), nil
+	}
+
+	if in.IsNull() {
+		return NewFloat32Null(), nil
+	}
+
+	var bigF *big.Float
+	err := in.As(&bigF)
+
+	if err != nil {
+		return nil, err
+	}
+
+	f, accuracy := bigF.Float32()
+	logging.FrameworkDebug(ctx, fmt.Sprintf("Float32Type ValueFromTerraform Float32 value %f, accuracy %d", f, accuracy))
+	f64, f64accuracy := bigF.Float64()
+	logging.FrameworkDebug(ctx, fmt.Sprintf("Float32Type ValueFromTerraform Float64 value %f, accuracy %d", f64, f64accuracy))
```

Comment:
Is there a specific scenario you're looking to capture with these debug logs? I'm not necessarily against it, but this code could use a comment if there's a situation where this information would be useful.

Do we care only when the accuracy is changed? Maybe that could trigger the log rather than always logging